vertical-pod-autoscaler/updater/updater.go

```diff
@@ -0,0 +1,186 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/client-go/tools/cache"
+	apiv1 "k8s.io/kubernetes/pkg/api/v1"
+
+	kube_client "k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+	v1lister "k8s.io/kubernetes/pkg/client/listers/core/v1"
+	"time"
+
+	"fmt"
+	"github.com/golang/glog"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/autoscaler/vertical-pod-autoscaler/updater/apimock"
+	"k8s.io/autoscaler/vertical-pod-autoscaler/updater/eviction"
+	"k8s.io/autoscaler/vertical-pod-autoscaler/updater/priority"
+	"k8s.io/autoscaler/vertical-pod-autoscaler/updater/recommender"
+)
+
+// Updater performs updates on pods if recommended by Vertical Pod Autoscaler
+type Updater interface {
+	// RunOnce represents single iteration in the main-loop of Updater
+	RunOnce()
+}
+
+type updater struct {
+	vpaLister        apimock.VerticalPodAutoscalerLister // wait for VPA api
+	podLister        v1lister.PodLister
+	recommender      recommender.CachingRecommender
+	evictionFactrory eviction.PodsEvictionRestrictionFactory
+}
+
+// RunOnce represents single iteration in the main-loop of Updater
+func (u *updater) RunOnce() {
+	vpaList, err := u.vpaLister.List()
+	if err != nil {
+		glog.Fatalf("failed get vpa list: %v", err)
+	}
+
+	if len(vpaList) == 0 {
+		glog.Warningf("no VPA objects to process")
+		return
+	}
+
+	for _, vpa := range vpaList {
+		glog.Infof("processing VPA object targetting %v", vpa.Spec.Target.MatchLabels)
```

Comment:
With all info logs specify the level `glog.V(2).Infof(...)`. Rule of thumb:
* V(1) - important stuff
* V(2) - slightly less important
* V(4) - per pod/per vpa logs.